catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -3508,4 +3508,14 @@ private boolean isMetacardBlacklisted(Metacard metacard) {
 
         return CollectionUtils.containsAny(tags, fanoutTagBlacklist);
     }
+
+    private ddf.security.Subject retrieveSubject() {
+        ddf.security.Subject subjectToReturn = null;
+        Subject subject = SecurityUtils.getSubject();
```

Comment:
Can you wrap this call in a try/catch for IllegalStateException and if this is thrown, write a debug log message.